src/core/libmaven/groupFiltering.cpp

```diff
@@ -64,7 +64,7 @@ bool GroupFiltering::filterByMS1(PeakGroup &peakgroup)
         return true;
 
     if (compound)
-        peakgroup.compound = compound;
+        peakgroup.setCompound(compound);
```

Comment:
@saifulbkhan Compound assignment should happen in Group stats or when the group is created. Not in filtering. I know this is not directly related to this PR so we can do this in a separate branch too, just mentioning it here for reference.

Reply:
wait, it's taking `_slice->compound` and assigning it again to `_slice` in `setPeakGroup`

Reply:
Ah correct. Will do.

Reply:
This still needs to be removed.